public/app/plugins/panel/alertlist/AlertInstances.tsx

```diff
@@ -48,6 +64,47 @@ export const AlertInstances = ({ alerts, options }: Props) => {
     }
   }, [filteredAlerts]);
 
+  const onShowAllClick = async () => {
+    if (!handleInstancesLimit) {
+      return;
+    }
+    await handleInstancesLimit(false);
+    setDisplayInstances(true);
+  };
+
+  const onShowLimitedClick = async () => {
+    if (!handleInstancesLimit) {
+      return;
+    }
+    await handleInstancesLimit(true);
+    setDisplayInstances(true);
+  };
+  const totalInstancesNumber = limitInstances ? grafanaFilteredInstancesTotal : filteredAlerts.length;
+  const limitStatus = limitInstances
+    ? `Showing ${INSTANCES_DISPLAY_LIMIT} of ${grafanaTotalInstances} instances`
+    : `Showing all ${grafanaTotalInstances} instances`;
+
+  const limitButtonLabel = limitInstances ? 'Remove limit' : `Limit the result to ${INSTANCES_DISPLAY_LIMIT} instances`;
```

Comment:
I think `Remove limit` is a bit too technical – can we use `View all instances` instead? Similar to what the alert list view shows:

<img width="280" alt="image" src="https://user-images.githubusercontent.com/868844/233991902-831955e3-28fb-4924-ad88-0352d9340f8e.png">